radix.go

```diff
@@ -142,6 +142,29 @@
 //
 // Use the golang.org/x/xerrors package if you're using an older version of go.
 //
+// Implicit pipelining
+//
+// Implicit pipelining is an optimization that reduces the overhead of
```

Comment:
I try to keep the main package doc very high level, and stick to only the bare minimum info that someone needs in order to use radix, or to preemptively answer questions that we get a lot (e.g. about AUTH). So I think these three paragraphs are a bit too wordy, but could be rolled together into one, like this:

> Implicit pipelining is an optimization implemented and enabled in the default Pool implementation (and therefore also used by Cluster and Sentinel) which involves delaying concurrent Cmds and FlatCmds a small amount of time and sending them to redis in a single batch, similar to manually using a Pipeline. By doing this radix significantly reduces the I/O and CPU overhead for concurrent requests.

> Note that only commands which do not block are eligible for implicit pipelining.

> See the documentation on Pool etc etc

What do you think?

Reply:
Sounds good to me. I updated the branch.